engines/crab/ui/KeyBindMenu.cpp

```diff
@@ -0,0 +1,242 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/*
+ * This code is based on the CRAB engine
+ *
+ * Copyright (c) Arvind Raja Yadav
+ *
+ * Licensed under MIT
+ *
+ */
+
+#include "crab/crab.h"
+#include "crab/ui/KeyBindMenu.h"
+
+namespace Crab {
+
+using namespace pyrodactyl::ui;
+using namespace pyrodactyl::image;
+using namespace pyrodactyl::text;
+using namespace pyrodactyl::input;
+
+void KeyBindMenu::load(rapidxml::xml_node<char> *node) {
+	if (nodeValid(node)) {
+		if (nodeValid("menu", node)) {
+			rapidxml::xml_node<char> *menode = node->first_node("menu");
+
+			if (nodeValid("primary", menode))
+				_prim.load(menode->first_node("primary"));
+
+			if (nodeValid("alt", menode))
+				_alt.load(menode->first_node("alt"));
+
+			if (nodeValid("prompt", menode))
+				_prompt.load(menode->first_node("prompt"));
+
+			if (nodeValid("inc", menode))
+				_inc.load(menode->first_node("inc"));
+
+			if (nodeValid("dim", menode))
+				_dim.load(menode->first_node("dim"));
+
+			if (nodeValid("divide", menode))
+				_divide.load(menode->first_node("divide"));
+
+			if (nodeValid("desc", menode))
+				_desc.load(menode->first_node("desc"));
+		}
+
+		// Initialize the menus
+		initMenu(CON_GAME);
+		initMenu(CON_UI);
+
+		if (nodeValid("controls", node))
+			_selControls.load(node->first_node("controls"));
+	}
+}
+
+void KeyBindMenu::startAndSize(const int &type, int &start, int &size) {
+	switch (type) {
+	case CON_GAME:
+		start = IG_START;
+		size = IG_SIZE;
+		break;
+	case CON_UI:
+		start = IU_START;
+		size = IU_SIZE;
+		break;
+	default:
+		break;
+	}
+}
+
+void KeyBindMenu::initMenu(const int &type) {
+	warning("STUB: KeyBindMenu::initMenu()");
```

Comment:
As I understand, you rely on our keymapper. Should this be cleaned up, then? Or do you plan to connect them together, so they stay in sync?